Gordon360/Services/HousingService.cs

```diff
@@ -531,4 +557,228 @@ public bool ChangeApplicationDateSubmitted(int applicationID)
         context.SaveChanges();
         return true;
     }
-}
+
+    /// <summary>
+    /// update roommate imformation
+    /// </summary>
+    /// <param name="username"> The username for the user who complete the aplication form </param>
+    /// <param name="application_id"> The ID of this application </param>
+    /// <param name="emailList"> A list of applicants' emails </param>
+    public async Task UpdateRoommateAsync(string username, string application_id, string[] emailList)
+    {
+        var account = accountService.GetAccountByUsername(username);
+        string gender = context.Student.FirstOrDefault(a => a.ID == account.GordonID).Gender;
+
+        var maxYear = context.Student
+            .Where(s => emailList.Contains(s.Email))
+            .Select(s => s.Class).Max();
+        await context.Year.AddAsync(new Year
+        {
+            ApplicationID = application_id,
+            Year1 = maxYear
+        });
+
+        foreach (string e in emailList)
+        {
+            if (e != "")
+            {
+                var existActiveApplicant = context.Applicant.FirstOrDefault(x => (x.Applicant1 == e) && (x.Active == 1));
+                if (existActiveApplicant != null)
+                {
+                    existActiveApplicant.Active = 0;
+                }
+                var student = context.Student.FirstOrDefault(x => x.Email == e);
+                if (student == null)
+                {
+                    throw new ResourceNotFoundException() { ExceptionMessage = "The applicant cannot be found." };
+                }
+                if (gender != student.Gender)
+                {
+                    throw new BadInputException() { ExceptionMessage = "The applicants are not of the same gender." };
+                }
+                gender = student.Gender;
+                var newApplicant = new Applicant
+                {
+                    ApplicationID = application_id,
+                    Applicant1 = e,
+                    Active = 1
+                }; ;
+                await context.Applicant.AddAsync(newApplicant);
+            }
+        }
```

Comment:
Would there be a way to simplify the code here?

Reply:
The logic seems fairly simple and straightforward to me. I'm not sure it needs to change.